net/src/Substrate.Gear.Client/Model/Types/Base/BaseNonZero.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using Substrate.NetApi.Model.Types.Base;
+
+namespace Substrate.Gear.Client.Model.Types.Base;
+
+public class BaseNonZero<T> : BaseType
```

Comment:
```suggestion
public abstract class BaseNonZero<T> : BasePrim<T>
```

Reply:
Possible, but needs refactoring with tests, may be later